### PR TITLE
feat(integration): Add support for Sanic >=21.3

### PR DESCRIPTION
**Pull request by ahmedetefy**
This PR:
- Adds support for latest Sanic versions >= 21.3 because it was initially broken due to breaking changes after LTS of 20.12

Note/Todo:
- There **MIGHT** be missing support for new added functionality in version 21.3 that should be integrated in future work
- This PR is merely there to not break support for what we already have

Fixes: #1066, #1128

### CODE REVIEW

File: sentry_sdk/integrations/sanic.py [hunk 1]
```diff
@@ -96,14 +96,21 @@ async def sentry_handle_request(self, request, *args, **kwargs):
...
-        def sentry_router_get(self, request):
-            # type: (Any, Request) -> Any
-            rv = old_router_get(self, request)
+        def sentry_router_get(self, *args):
+            # type: (Any, Union[Any, Request]) -> Any
+            if version >= (21, 3):
+                rv = old_router_get(self, *args)
+            else:
+                request = args[0]
+                rv = old_router_get(self, request)
```
Comment (untitaker) [inline sentry_sdk/integrations/sanic.py:105]: not sure why this branch is necessary, why not always do `*args`?

File: tox.ini [hunk 2]
```diff
@@ -139,6 +139,9 @@ deps =
     sanic-0.8: sanic>=0.8,<0.9
     sanic-18: sanic>=18.0,<19.0
     sanic-19: sanic>=19.0,<20.0
+    sanic-20: sanic>=20.0,<21.0
+    {py3.7,py3.8,py3.9}-sanic-21: sanic>=21.0
```
Comment (untitaker) [inline tox.ini:143]: You haven't declared those jobs to run in line 40. If you run `tox -l` there will not be an entry for sanic 20 and sanic 21.

**Review by untitaker — CHANGES_REQUESTED** [review]

File: sentry_sdk/integrations/sanic.py (hunk 1 above)
Comment (ahmedetefy) [reply sentry_sdk/integrations/sanic.py]: Yeah I agree.. added it initially to make the distinction but I guess no need

File: tests/integrations/sanic/test_sanic.py [hunk 3]
```diff
@@ -34,7 +40,11 @@ def test_request_data(sentry_init, app, capture_events):
     assert response.status == 200
 
     (event,) = events
-    assert event["transaction"] == "hi"
+    # Sanic versions above and including 21.3 append the app name to the route name
```
Comment (AbhiPrasad) [inline tests/integrations/sanic/test_sanic.py:43]: Should this be the default? I think we should strip the app name so transactions stay consistent in between updates to the Sanic integration.

File: tox.ini [hunk 4]
```diff
@@ -139,6 +141,9 @@ deps =
     sanic-0.8: sanic>=0.8,<0.9
     sanic-18: sanic>=18.0,<19.0
     sanic-19: sanic>=19.0,<20.0
+    sanic-20: sanic>=20.0,<21.0
+    sanic-21: sanic>=21.0
```
Comment (AbhiPrasad) [inline tox.ini:145]: This might break CI with sanic 22, is that fine?

File: tests/integrations/sanic/test_sanic.py (hunk 3 above)
Comment (ahmedetefy) [reply tests/integrations/sanic/test_sanic.py]: you are right! resolved

File: tox.ini (hunk 4 above)
Comment (ahmedetefy) [reply tox.ini]: oops

File: sentry_sdk/integrations/sanic.py [hunk 5]
```diff
@@ -96,14 +96,23 @@ async def sentry_handle_request(self, request, *args, **kwargs):
...
                     with hub.configure_scope() as scope:
-                        scope.transaction = rv[0].__name__
+                        if version >= (21, 3):
+                            # Sanic versions above and including 21.3 append the app name to the
+                            # route name, and so we need to remove it from Route name so the
+                            # transaction name is consistent across all versions
+                            sanic_app_name = self.ctx.app.name
+                            scope.transaction = rv[0].name.split(
+                                "%s." % sanic_app_name
+                            )[1]
```
Comment (AbhiPrasad) [inline sentry_sdk/integrations/sanic.py:113]: Could the name.split list length ever be 0 or 1?

Comment (ahmedetefy) [reply sentry_sdk/integrations/sanic.py:113]: I do not think so https://github.com/sanic-org/sanic/blob/108a4a99c7e12561b35b62e26ecfc33c77db802d/sanic/mixins/routes.py#L666

Comment (ahmedetefy) [reply sentry_sdk/integrations/sanic.py:113]: `self.name` cannot be null

Comment (ahmedetefy) [reply sentry_sdk/integrations/sanic.py:113]: I will add safe guards regardless

**Review by AbhiPrasad — APPROVED** [review]
Let's ship to unblock users.

**Pull request merged**